--- a/media/base/media_channel.h
+++ b/media/base/media_channel.h
@@ -62,6 +62,10 @@
 class VideoFrame;
 }  // namespace webrtc

+namespace webrtc {
+  class RtpPacketSinkInterface;
+}
+
 namespace cricket {

 class AudioSource;
@@ -245,6 +249,9 @@
       uint32_t ssrc,
       rtc::scoped_refptr<webrtc::FrameDecryptorInterface> frame_decryptor);

+  // Sets a sink for raw RTP packets.
+  virtual void SetRawRtpPacketSink(webrtc::RtpPacketSinkInterface* sink) {}
+
   // Enable network condition based codec switching.
   virtual void SetVideoCodecSwitchingEnabled(bool enabled);

--- a/pc/channel_manager.cc
+++ b/pc/channel_manager.cc
@@ -45,6 +45,26 @@
   worker_thread_->Invoke<void>(RTC_FROM_HERE, [&] { media_engine_.reset(); });
 }

+MediaChannel* ChannelManager::GetVoiceChannel(
+    const std::string& transport_name) {
+  for (const auto& channel : voice_channels_) {
+    if (channel->transport_name() == transport_name) {
+      return channel.get();
+    }
+  }
+  return nullptr;
+}
+
+MediaChannel* ChannelManager::GetVideoChannel(
+    const std::string& transport_name) {
+  for (const auto& channel : video_channels_) {
+    if (channel->transport_name() == transport_name) {
+      return channel.get();
+    }
+  }
+  return nullptr;
+}
+
 bool ChannelManager::SetVideoRtxEnabled(bool enable) {
   // To be safe, this call is only allowed before initialization. Apps like
   // Flute only have a singleton ChannelManager and we don't want this flag to
--- a/pc/channel_manager.h
+++ b/pc/channel_manager.h
@@ -88,8 +88,9 @@
   // Shuts down the media engine.
   void Terminate();

-  // The operations below all occur on the worker thread.
-  // ChannelManager retains ownership of the created channels, so clients should
+  // Getters for existing channels, returning the base MediaChannel type.
+  MediaChannel* GetVoiceChannel(const std::string& transport_name);
+  MediaChannel* GetVideoChannel(const std::string& transport_name);
+
   // call the appropriate Destroy*Channel method when done.
   VoiceChannel* CreateVoiceChannel(
       webrtc::Call* call, const cricket::MediaConfig& media_config,
--- a/pc/peer_connection.h
+++ b/pc/peer_connection.h
@@ -112,6 +112,9 @@
     MAX_VALUE = 0x8000,
   };

+  // Make channel_manager() public.
+  cricket::ChannelManager* channel_manager() const;
+
   explicit PeerConnection(PeerConnectionFactory* factory,
                           std::unique_ptr<RtpEventLog> event_log,
                           std::unique_ptr<Call> call);
@@ -926,8 +929,6 @@
   // Returns RTCError::OK() if there are no issues.
   RTCError ValidateConfiguration(const RTCConfiguration& config) const;

-  cricket::ChannelManager* channel_manager() const;
-
   enum class SessionError {
     kNone,       // No error.
     kContent,    // Error in BaseChannel SetLocalContent/SetRemoteContent.

